spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8ConfigDataLocationResolver.java

```diff
@@ -51,91 +51,52 @@ public Fabric8ConfigDataLocationResolver(DeferredLogFactory factory) {
-						.registerSingleton("configDataConfigMapPropertySourceLocator",
-								event.getBootstrapContext().get(ConfigMapPropertySourceLocator.class)));
-			}
-			if (secretsProperties != null && secretsProperties.enabled()) {
-				Fabric8SecretsPropertySourceLocator secretsPropertySourceLocator = new Fabric8SecretsPropertySourceLocator(
-						kubernetesClient, secretsProperties, namespaceProvider);
-				bootstrapContext.registerIfAbsent(SecretsPropertySourceLocator.class,
-						InstanceSupplier.of(secretsPropertySourceLocator));
-				bootstrapContext.addCloseListener(event -> event.getApplicationContext().getBeanFactory()
-						.registerSingleton("configDataSecretsPropertySourceLocator",
-								event.getBootstrapContext().get(SecretsPropertySourceLocator.class)));
-			}
-		}
-	}
+		KubernetesClient kubernetesClient = registerConfigAndClient(bootstrapContext, kubernetesClientProperties);
 
-	private void registerRetryBeans(ConfigMapConfigProperties configMapProperties,
-			SecretsConfigProperties secretsProperties, ConfigurableBootstrapContext bootstrapContext,
-			KubernetesClient kubernetesClient, KubernetesNamespaceProvider namespaceProvider) {
 		if (configMapProperties != null && configMapProperties.enabled()) {
 			ConfigMapPropertySourceLocator configMapPropertySourceLocator = new Fabric8ConfigMapPropertySourceLocator(
-					kubernetesClient, configMapProperties, namespaceProvider);
+				kubernetesClient, configMapProperties, namespaceProvider);
 			if (isRetryEnabledForConfigMap(configMapProperties)) {
 				configMapPropertySourceLocator = new ConfigDataRetryableConfigMapPropertySourceLocator(
-						configMapPropertySourceLocator, configMapProperties);
+					configMapPropertySourceLocator, configMapProperties);
 			}
 
-			bootstrapContext.registerIfAbsent(ConfigMapPropertySourceLocator.class,
-					InstanceSupplier.of(configMapPropertySourceLocator));
-			bootstrapContext.addCloseListener(event -> event.getApplicationContext().getBeanFactory().registerSingleton(
-					"configDataConfigMapPropertySourceLocator",
-					event.getBootstrapContext().get(ConfigMapPropertySourceLocator.class)));
+			registerSingle(bootstrapContext, ConfigMapPropertySourceLocator.class,
+				configMapPropertySourceLocator, "configDataConfigMapPropertySourceLocator");
 		}
 
 		if (secretsProperties != null && secretsProperties.enabled()) {
 			SecretsPropertySourceLocator secretsPropertySourceLocator = new Fabric8SecretsPropertySourceLocator(
-					kubernetesClient, secretsProperties, namespaceProvider);
+				kubernetesClient, secretsProperties, namespaceProvider);
 			if (isRetryEnabledForSecrets(secretsProperties)) {
 				secretsPropertySourceLocator = new ConfigDataRetryableSecretsPropertySourceLocator(
-						secretsPropertySourceLocator, secretsProperties);
+					secretsPropertySourceLocator, secretsProperties);
 			}
-			bootstrapContext.registerIfAbsent(SecretsPropertySourceLocator.class,
```

Comment:
all  this code in the form:

```
bootstrapContext.registerIfAbsent(SecretsPropertySourceLocator.class,
					InstanceSupplier.of(secretsPropertySourceLocator));
			bootstrapContext.addCloseListener(event -> event.getApplicationContext().getBeanFactory().registerSingleton(
					"configDataSecretsPropertySourceLocator",
					event.getBootstrapContext().get(SecretsPropertySourceLocator.class)));
```

was replaced with `registerSingle` that already existed from one of my previous PRs